### ISSUE TITLE
Weird progress bar issue (needs more investigation)

### ISSUE THREAD

**Issue by benwiley4000** [benwiley4000/cassette#310]
Issue reported originally by @danielr18 in #305:

> I have a weird bug that isn't really related to seek, but the ProgressBar component. 
![image](https://user-images.githubusercontent.com/9046616/48921418-6bff8d00-ee7e-11e8-8371-a07f76e20ca9.png)
>
> Somehow when I click ~2 times over the seek buttons, red squares in the image, the player seeks to the position above of where I clicked. If you want, you can try to recreate it here: http://54.224.96.152:3001/, perhaps it's a bug we can solve in the library?

This is possibly related to #297 (still pretty unclear). We need some more investigation .. still waiting on source code for a minimal repro from @danielr18.

**Comment by danielr18** [comment]
Here' s the minimal repro:
https://codesandbox.io/s/w7kp542vw7

**Comment by benwiley4000** [comment]
Thanks so much!

**Comment by benwiley4000** [comment]
@danielr18 I've got a solution that solves your issue.

Where you have:

```jsx
    onSeekPreview(newCurrentTime);
    setTimeout(onSeekComplete, 100);
```

you can do:

```jsx
    onSeekComplete(newCurrentTime);
```

and it fixes your issue.

The reason why is that the `mousemove` event is sometimes triggered while clicking the forward/back buttons and it fires in the 100ms timeout between `onSeekPreview` and `onSeekComplete`. The `ProgressBar` component, meanwhile, relies on the `adjusting` boolean prop which tells it whether a progress adjustment is in progress, and this is based on the `seekInProgress` state from the `playerContext`.

Although the change I suggested will fix your specific problem, it's still an issue that any `MediaProgressBar` rendered on screen will update in response to a `mousemove` event if media progress is being adjusted anywhere. I think this may also be the reason for #297.

The real solution is to move that `adjusting` state to be internal to the `ProgressBar` component, rather than passed in as a prop.

**Comment by danielr18** [comment]
> @danielr18 I've got a solution that solves your issue.
> 
> Where you have:
> 
> ```js-jsx
>     onSeekPreview(newCurrentTime);
>     setTimeout(onSeekComplete, 100);
> ```
> 
> you can do:
> 
> ```js-jsx
>     onSeekComplete(newCurrentTime);
> ```
> 
> and it fixes your issue.

Ohh, interesting, I had that as the temporary solution before we worked on this PR #309.

**Comment by benwiley4000** [comment]
The fix for this is out in v2.0.0-alpha.8! :) https://github.com/benwiley4000/cassette/releases/tag/v2.0.0-alpha.8
